src/parser/Tokenizer.h

```diff
@@ -16,16 +16,17 @@ using namespace std::string_literals;
 
 /// Helper function for ctre: concatenation of fixed_strings
 template <size_t A, size_t B>
-constexpr ctll::fixed_string<A + B - 1> operator+(
-    const ctll::fixed_string<A>& a, const ctll::fixed_string<B>& b) {
-  char32_t comb[A + B - 1] = {};
-  for (size_t i = 0; i < A - 1;
-       ++i) {  // omit the trailing 0 of the first string
+constexpr ctll::fixed_string<A + B> operator+(const ctll::fixed_string<A>& a,
+                                              const ctll::fixed_string<B>& b) {
+  char32_t comb[A + B + 1] = {};
+  for (size_t i = 0; i < A; ++i) {  // omit the trailing 0 of the first string
     comb[i] = a.content[i];
   }
   for (size_t i = 0; i < B; ++i) {
-    comb[i + A - 1] = b.content[i];
+    comb[i + A] = b.content[i];
   }
+  // the input array must be zero-terminated
+  comb[A + B] = '\0';
```

Comment:
1. Was the 0 part of the string length before and now it isn't?
2. Do we use `++i` consistently instead of `i++` ? I am aware that there is a difference between the two, when the value is used. Is there a difference when it is not used?

Reply:
1. Yes exactly, in the new `ctre` version "a" becomes `fixed_string<1>` (previously `fixed_string<2>`). We have exhaustive tests that would horribly break if this method was wrong.

2. If the value is used, then post- vs. pre-increment makes a huge difference, as you have correctly stated (has been like this "forever" in C. When the value is not used, then `++i` is never less efficient than `i++`. For builtin types, like `int` etc. it is exactly the same. But if the incremented object `i` is e.g. an iterator to a complex data structure, which stores several pointers etc., then `i++` has to create a copy of the iterator (we need to old value to possibly still use it, and the new value, because we requested an increment. For this reason, I have started to use `++i` as a default in for loops (of course only where range-based for loops don't apply.